fotoapparat/src/main/java/io/fotoapparat/parameter/range/Range.java

```diff
@@ -0,0 +1,36 @@
+package io.fotoapparat.parameter.range;
+
+/**
+ * Created by ychen on 5/25/2017.
+ */
+
+public abstract class Range<T> {
+    abstract public boolean contains(T value);
+    abstract public T highest();
+    abstract public T lowest();
+
+    public String toString() {
```

Comment:
I would not put it here, because it could be that `lowest()` and `highest()` implementations might perform costly operations. So, I would instead generate different `toString` implementation for `DiscreteRange` and for `ContinuosRange`.